Sources/BraveWallet/Crypto/Stores/NFTStore.swift

```diff
@@ -235,100 +270,176 @@ public class NFTStore: ObservableObject, WalletObserverStore {
         simpleHashSpamNFTs: simpleHashSpamNFTs
       )
       
-      let allNetworkAssets = userVisibleAssets + userHiddenAssets + unionedSpamNFTs
-      userNFTs = buildAssetViewModels(allUserAssets: allNetworkAssets)
-        .optionallyFilterUnownedNFTs(
-          isHidingUnownedNFTs: filters.isHidingUnownedNFTs,
-          selectedAccounts: selectedAccounts
-        )
+      var allNetworkAssets: [NetworkAssets] = []
+      for networkAssets in userVisibleAssets {
+        let hiddenAssets = userHiddenAssets.first(where: {
+          $0.network.chainId == networkAssets.network.chainId && $0.network.coin == networkAssets.network.coin
+        })?.tokens ?? []
+        let spamNFTs = unionedSpamNFTs.first(where: {
+          $0.network.chainId == networkAssets.network.chainId && $0.network.coin == networkAssets.network.coin
+        })?.tokens ?? []
+        allNetworkAssets.append(.init(network: networkAssets.network, tokens: networkAssets.tokens + hiddenAssets + spamNFTs, sortOrder: networkAssets.sortOrder))
+      }
```

Comment:
This logic (including fetching `userVisibleAssets`, `userHiddenAssets`, & `simpleHashSpamNFTs` above the fold here) is duplicated in `updateNFTStatus(_:visible:isSpam:isDeletedByUser:)` below, can it be shared? 🤔

Maybe it could be split into a separate function that they both share, I think this `updateNFTStatus()` function is trying to avoid network requests so we can't simply call `update()`.